lib/Onnxifi/Base.cpp

```diff
@@ -60,20 +60,46 @@ onnxStatus Graph::initGraph(const void *onnxModel, size_t onnxModelSize,
     return ONNXIFI_STATUS_INTERNAL_ERROR;
   }
 
-  onnxNameToInputVar_ = loader->getInputVarsMapping();
-  onnxNameToOutputNode_ = loader->getOutputVarsMapping();
+  onnxInputToPlaceholder_ = loader->getInputVarsMapping();
+  onnxOutputToPlaceholder_ = loader->getOutputVarsMapping();
 
   // Emit IR for the graph and compile it.
   backendPtr_->getEE().compile(CompilationMode::Infer, function_, ctx_);
 
   return ONNXIFI_STATUS_SUCCESS;
 }
 
-onnxStatus Graph::run() {
+void Graph::runAsync(EventPtr inputEvent, EventPtr outputEvent) {
+  llvm::DenseMap<Placeholder *, onnxPointer> inputPlaceholderToBuffer =
+      inputPlaceholderToBuffer_;
+  llvm::DenseMap<Placeholder *, onnxPointer> outputPlaceholderToBuffer =
+      outputPlaceholderToBuffer_;
+
+  // Once inputs are copied we can allow processing concurrent requests.
+  inputRunMutex_.unlock();
+
+  // Submit graph for asynchronous execution.
+  // Concurrency for executing 'run' method is limited by number of threads in
+  // the backend specific thread pool.
+  backend()->runAsync([inputEvent, outputEvent, inputPlaceholderToBuffer,
```

Comment:
Is it good to capture these maps by value?

Reply:
That's copied intentionally so that those maps internal to graph class are not shared.